lib/auth/auth.go

```diff
@@ -3053,7 +3063,11 @@ func (a *Server) isMFARequired(ctx context.Context, checker services.AccessCheck
 		if db == nil {
 			return nil, trace.Wrap(notFoundErr)
 		}
-		noMFAAccessErr = checker.CheckAccessToDatabase(db, services.AccessMFAParams{AlwaysRequired: false, Verified: false})
+		noMFAAccessErr = checker.CheckAccess(
+			db,
+			services.AccessMFAParams{AlwaysRequired: false, Verified: false},
+			nil, // we do not check labels for databases
```

Comment:
This is the only place we didn't. Other uses of `CheckAccessToDatabase` all passed a `DatabaseLabelMatcher`, so we're just preserving the original behavior here.

Reply:
@r0mant could you confirm that this is intentional?

Reply:
I believe this to be intentional. This code is attempting to determine if MFA is required by running the check (without label matching) and seeing if the error that comes back is `ErrSessionMFARequired`.

If we do match labels, and the RBAC check fails due to labels, then we'll return access denied (before even considering MFA) instead of `ErrSessionMFARequired` and this function will report that MFA is not required.

That said, the same checks for MFA being enabled on nodes and k8s clusters _do_ match labels.

Reply:
I don't know, I didn't write this particular piece code but it seems like an omission to me. The labels will be checked by the database agent anyway but we could do it here too for consistency. Seems like if the labels don't match, there's no point in checking MFA either.

Reply:
Ok cool, I will remove the argument entirely and always check labels. Seems like what we'd want anyway since labels are core to our whole RBAC system.